hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java

```diff
@@ -139,10 +146,126 @@ void nonLazyCreateCandidateMustHaveReceiver() {
         assertAbortsWith(wellKnownRelayedHapiCall(0), INVALID_CONTRACT_ID);
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    void ethCreateHappyPathAsExpected() {
+        final var inOrder = inOrder(worldUpdater, frameBuilder, frameRunner, gasCharging, senderAccount);
+
+        givenSenderAccount();
+        givenRelayerAccount();
+
+        final var context = wellKnownContextWith(code, blocks);
+        final var transaction = wellKnownRelayedHapiCreate();
+
+        given(gasCharging.chargeForGas(senderAccount, relayerAccount, context, worldUpdater, transaction))
+                .willReturn(CHARGING_RESULT);
+        given(senderAccount.getAddress()).willReturn(EIP_1014_ADDRESS);
+        given(senderAccount.getNonce()).willReturn(NONCE);
+        final var expectedToAddress = Address.contractAddress(EIP_1014_ADDRESS, NONCE);
+        given(frameBuilder.buildInitialFrameWith(
+                        transaction,
+                        worldUpdater,
+                        context,
+                        config,
+                        EIP_1014_ADDRESS,
+                        expectedToAddress,
+                        CHARGING_RESULT.intrinsicGas()))
+                .willReturn(initialFrame);
+        given(frameRunner.runToCompletion(
+                        transaction.gasLimit(), initialFrame, tracer, messageCallProcessor, contractCreationProcessor))
+                .willReturn(SUCCESS_RESULT);
+        given(initialFrame.getSelfDestructs()).willReturn(Set.of(NON_SYSTEM_LONG_ZERO_ADDRESS));
+
+        final var result =
+                subject.processTransaction(transaction, worldUpdater, () -> feesOnlyUpdater, context, tracer, config);
+
+        inOrder.verify(worldUpdater).setupAliasedCreate(Address.ZERO, expectedToAddress);
```

Comment:
Just out of curiosity, is the "in order" verification of calls part of the behavior specification?  I mean, does it matter that `senderAccount.incrementNonce` happens _after_ `worldUpdater.setupAliasedCreate` and _before_ `gasCharging.chargeForGas` as long as all three get done?

Reply:
The order of `incrementNonce()` and `setupAliasedCreate()` do matter for a `CONTRACT_CREATION`, since the sender nonce affects the `CREATE1` address calculation.